### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false positive for `stroke-opacity`

### ISSUE THREAD

**Issue by Titoubiz** [stylelint/stylelint#8080]
### What minimal example or steps are needed to reproduce the bug?

```html
<svg>
    <line />
</svg>

<style>
    line {
        stroke-opacity: 50%;
    }
</style>
```

### What minimal configuration is needed to reproduce the bug?

```json
{
    "overrides": [
        { 
            "customSyntax": "postcss-html",
            "files": ["*.html"] 
        }
    ],
    "rules": {
        "declaration-property-value-no-unknown": true
    }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIA8BnAbgcwHwB0B2OALkgDYCWeMABAPS55I3rb76qECeJM9R5lVYK0JEUhAE4QA1jAC0EAA4BDMGU7wqAVgAMAUgDcwgL5smnblhAAaEADMy3AHJKAtnEQVYADwB0AC0IXEmtwCDx7DAQQITwqKhwQCDQYcXEyWBQEjQBtfDi44HjY-LiEsABXMQgXAGUOPEIlLyz4kAUIMTAUFFkAoISrPJLW+25MhCpckAAqf0CSBIBdIpKTYsXB4oTxcrGWmJKE2DASJXElQjIw2QVJBRTOWTQlEnK5PAhZcrwpD4B3AgTCRvIZrIwhSDhMgYABiEHELguUQAVigwiFYAoUFEDq0xFwYPwRBMEqdCDAxAMhgl2p1ur15i1SRcKcT8OCjEA)

### Which Stylelint-related dependencies are you using?

```json
{
    "postcss-html": "^1.7.0",
    "stylelint": "^16.10.0"
}
```

### What did you expect to happen?

```
✅ No problems!
```

### What actually happened?

```
7:19-22    ERROR    Unexpected unknown value "50%" for property "stroke-opacity"
```

**Comment by Mouvedia** [comment]
Unless https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-opacity is incorrect, it seems that the error is accurate.
i.e. it should either be `0` or `1`

**Comment by Mouvedia** [comment]
It seems like `css-tree` uses `<number-zero-one>` which reject percentages.

**Comment by Mouvedia** [comment]
@Titoubiz we may choose to add temporary fixes on our side if `css-tree` doesn't merge PRs in a _timely_ manner.
No established time span has been chosen yet but if we eventually had to, it would be part of a batch.
i.e. alongside #8027, #7241 and #7964

**Comment by jeddy3** [comment]
@Titoubiz Thanks for the report and for using the template.

As @Mouvedia said, it's an upstream issue. However, you have two options in the meantime:

- use a number rather than a percentage; this is the convention our standard config promotes ([demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgZxwCcIBrGAWggAckwBLHAT3hQFYAGAUgG4sBfLCAA0IAGZ0ANjABySALZxEMAB7yqUgHRg8eYeAiZxAcwQgMmFCnQgVOGJih5rLANpZLl6wUZSJdTDjkkIZ0RuQESA5IhFDW7igAuvx6wcYAYhCEckg4pgBWeAZ6sFS6iOaeIN6+-jjOViASOTAE1kLxXkw1AUEGxuE4kVDRsQgNTXatIMl8QA))
- extend `declaration-property-value-no-unknown`'s syntax to allow percentages:

```json
{
    "rules": {
      "declaration-property-value-no-unknown": [true, {
        "propertiesSyntax": { "stroke-opacity": "|<percentage>" }
      }]
    }
}
```
[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgZxwCcIBrGAWggAckwBLHAT3hQFYAGAUgG4sBfLCAA0IAGZ0ANjABySALZxEMAB7yqUgHRg8eYeAiZxAcwQgMmFJZToQhAK5S8NluatWbsMBKSEkOOgbkVMRUMIRM5ABuSBJ2FJgQ5HaYJAkA7tgIKADaRHFCaFhubjbB1GH+MHgAyoyYOEjKzmjWIATEZJQ09EzNNgA+ADyhhGAw9UhGMAB8NigCFsV8ALpFVgt8epCGdEYAYhCEcn6mAFZ4BnqwVLqIrq0EjFISdPV9IN44VTg2Qms2j2erxw5G2xnIBCQmCgPig70+3xs-BAfCAA)

**Comment by ryo-manba** [comment]
I've opened an issue in csstree that includes a potential solution for this problem.
https://github.com/csstree/csstree/issues/293

**Comment by romainmenke** [comment]
Sorry, I don't think this is ready to implement.
I think this should remain blocked

**Comment by Mouvedia** [comment]
Then Ill explain why it should.
[`css-tree` 3.0.1 has been released](https://www.npmjs.com/package/css-tree/v/3.0.1) which contains [4175e97e1](https://github.com/csstree/csstree/commit/4175e97e17747cdd67bc84cfc6f3f6b6f02e7ea3).

The way @ryo-manba did it, is a bit too hasty though. He should have waited for the dependabot PR and added commits to it.

**Comment by romainmenke** [comment]
Ah, ok, I missed that there was a `csstree` update.

Yeah, in that case we would just trigger a dependabot update.
I also don't think we need to add extra test coverage for such changes.

**Comment by Mouvedia** [comment]
> I also don't think we need to add extra test coverage for such changes.

👍 

> Yeah, in that case we would just trigger a dependabot update.

I triggered a dependabot update earlier in the day; it wasn't successful.

**Comment by romainmenke** [comment]
Dependabot ran out of available PR's to open.
I've merged a few of the changes to make room.

**Comment by romainmenke** [comment]
See: https://github.com/stylelint/stylelint/pull/8105

**Comment by romainmenke** [comment]
Also wanted to say that I commented to quickly on this thread in https://github.com/stylelint/stylelint/issues/8080#issuecomment-2452927470

I had missed the csstree update and was concerned that this would be implemented as a manual patch within Stylelint itself.

I should have looked more carefully at the entire state and context before commenting :)

----

Thank you @ryo-manba for working on this issue in `mdn/data` and in `csstree`, really awesome work!
